src/Tasks/Microsoft.NET.Build.Tasks/PublishAssembliesResolver.cs

```diff
@@ -24,27 +25,38 @@ public PublishAssembliesResolver WithPrivateAssets(IEnumerable<string> privateAs
             _privateAssetPackageIds = privateAssetPackageIds;
             return this;
         }
+        public PublishAssembliesResolver PreserveCacheLayout(string cond)
+        {
+            _preserveCacheLayout = String.IsNullOrEmpty(cond) ? false : cond.ToLowerInvariant().Equals("true");
+            return this;
+        }
 
         public IEnumerable<ResolvedFile> Resolve(ProjectContext projectContext)
         {
             List<ResolvedFile> results = new List<ResolvedFile>();
 
             foreach (LockFileTargetLibrary targetLibrary in projectContext.GetRuntimeLibraries(_privateAssetPackageIds))
             {
-                string libraryPath = _packageResolver.GetPackageDirectory(targetLibrary.Name, targetLibrary.Version);
+                string pkgRoot;
+                string libraryPath = _packageResolver.GetPackageDirectory(targetLibrary.Name, targetLibrary.Version, out pkgRoot);
 
-                results.AddRange(GetResolvedFiles(targetLibrary.RuntimeAssemblies, libraryPath));
-                results.AddRange(GetResolvedFiles(targetLibrary.NativeLibraries, libraryPath));
+                results.AddRange(GetResolvedFiles(targetLibrary.RuntimeAssemblies, libraryPath, pkgRoot, "runtime"));
+                results.AddRange(GetResolvedFiles(targetLibrary.NativeLibraries, libraryPath, pkgRoot, "native"));
 
                 foreach (LockFileRuntimeTarget runtimeTarget in targetLibrary.RuntimeTargets.FilterPlaceHolderFiles())
                 {
                     if (string.Equals(runtimeTarget.AssetType, "native", StringComparison.OrdinalIgnoreCase) ||
                         string.Equals(runtimeTarget.AssetType, "runtime", StringComparison.OrdinalIgnoreCase))
                     {
+                        string srcpath = Path.Combine(libraryPath, runtimeTarget.Path);
```

Comment:
(nit) the casing/naming of this variable doesn't really match existing code.  It could be `srcPath` or `sourcePath`.